src/Metadata/Property/Factory/ConstructorPropertyMetadataFactory.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Metadata\Property\Factory;
+
+use ApiPlatform\Core\Api\OperationMethodResolverInterface;
+use ApiPlatform\Core\Exception\PropertyNotFoundException;
+use ApiPlatform\Core\Exception\RuntimeException;
+use ApiPlatform\Core\Metadata\Property\PropertyMetadata;
+
+/**
+ * Set properties available in the constructor as writable.
+ *
+ * @author Maxime Veber <maxime.veber@nekland.fr>
+ */
+final class ConstructorPropertyMetadataFactory implements PropertyMetadataFactoryInterface
+{
+    private $decorated;
+    private $operationMethodResolver;
+
+    public function __construct(OperationMethodResolverInterface $operationMethodResolver, PropertyMetadataFactoryInterface $decorated = null)
+    {
+        $this->decorated = $decorated;
+        $this->operationMethodResolver = $operationMethodResolver;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(string $resourceClass, string $property, array $options = []): PropertyMetadata
+    {
+        if (null === $this->decorated) {
+            $propertyMetadata = new PropertyMetadata();
+        } else {
+            try {
+                $propertyMetadata = $this->decorated->create($resourceClass, $property, $options);
+            } catch (PropertyNotFoundException $propertyNotFoundException) {
+                $propertyMetadata = new PropertyMetadata();
+            }
+        }
+
+        if (!isset($options['collection_operation_name'])) {
+            return $propertyMetadata;
+        }
+
+        // Constructor arguments are obviously accessible only on post operation, put will result in an error.
+        try {
+            $method = $this->operationMethodResolver->getCollectionOperationMethod($resourceClass, $options['collection_operation_name']);
+        } catch (RuntimeException $e) {
+            return $propertyMetadata;
+        }
+
+        if ('POST' !== $method) {
+            return $propertyMetadata;
+        }
+
+        $ref = new \ReflectionClass($resourceClass);
+        if (!$ref->isInstantiable() || !$constructor = $ref->getConstructor()) {
+            return $propertyMetadata;
+        }
+
+        foreach ($constructor->getParameters() as $constructorParameter) {
+            if ($constructorParameter->name === $property) {
```

Comment:
We must also check that the current value of `writable` is null (to avoid overwriting a manual setting tru `@ApiProperty`).

Reply:
Hum. I think modifying the order would be better because the property accessor will set this value to "false".

Actually, this issue is also present in the `ExtractorPropertyMetadataFactory`: if the property accessor decided to set the property to "not accessible" (aka `false`), no manual configuration can reverse the value because it fills it only if the current value is "null".

What do you think of that?

Reply:
The user's manual configuration must always be able to override anything. And yes, I think this decorator should run after the PropertyInfo decorator.

Reply:
Ok, so this answer means that there's an issue inside the `ExtractorPropertyMetadataFactory`.

It also means as I understand that I can indeed override the `writable` value (so I don't modify this line).